src/Marker.tsx

```diff
@@ -38,6 +40,17 @@ export class Marker extends React.Component<MarkerProps, object> {
   }
   // change the position automatically if the props get changed
   public componentWillReceiveProps(nextProps: MarkerProps) {
+    // Rerender the marker if child props change
+    if (React.Children.count(nextProps.children) > 0
+      && React.Children.count(this.props.children) > 0
+      && !isEqual(nextProps.children.props, this.props.children.props)) {
+      const { markersGroup } = this.context;
+      if (this.marker) {
+        markersGroup.removeObject(this.marker);
+      }
+      this.marker = this.renderChildren(nextProps.children, nextProps.lat, nextProps.lng)  
+      return
```

Comment:
The return is meaningless here please remove it,
There is no reason to stop code execution after that point as we might get out of sync between props and state.